### ISSUE TITLE
num_updates reported at validation time

### ISSUE THREAD

**Issue by newpro** [facebookresearch/ParlAI#2046]
## Bug description
I apologize in advance, I have not looked deep into the code. This is a quick bug report based on what I saw.

Bug: It is observed in logs that the model log "num_of_update" that is not zero. The bug makes me very hesitate to report the accuracy I am getting in my paper. For example, this is a line from testing log:

"222s elapsed: {'exs': 7300, '%done': '98.01%', 'time_left': '4s', 'accuracy': 0.4058, 'f1': 0.4307, 'hits@1': 0.406, 'hits@5': 0.719, 'hits@10': 0.88, 'hits@100': 1.0, 'bleu': 0.2901, 'lr': 5e-10, **'num_updates': 92797**, 'examples': 7300, 'loss': 949.2, 'mean_loss': 0.13, 'mean_rank': 4.308, 'mrr': 0.5503}"

The value should be zero, according to the definition of the testing, otherwise it would indicate during testing, the model has been updated. 

See line [here](https://github.com/facebookresearch/ParlAI/blob/a0fe44646be2e8064b476b347e09cc1b20121e4b/parlai/core/torch_agent.py#L1002) in report function.

`metrics['num_updates'] = self._number_training_updates`

## Reproduction steps
Performed fine-turn on my own dataset. The bug is observed during testing.

Notice that the bug is also expected, but not supposed to be, in the BERT Light model. See [here](https://github.com/facebookresearch/ParlAI/blob/56513cade9d9a0cffddfc9b731d2e898ba10af7f/parlai/zoo/model_list.py#L251) and [here](https://parl.ai/docs/zoo.html?highlight=num_updates#light-bert-biranker-dialogue-model) in docs.

## Expected behavior
In testing, the num_updates should = 0.

## Additional context
Apologize again. I have not look into the code much deeper. Sorry can not be much help, u got to dig a bit.

In my opinion. Two possible situations are muturally exclusive, and may happened in this case:
* The model is indeed updating during testing. Which is very bad. The log is printing correctly as training update. See line [here](https://github.com/facebookresearch/ParlAI/blob/a0fe44646be2e8064b476b347e09cc1b20121e4b/parlai/core/torch_agent.py#L1002).
* The model is not updating, but somehow logs the num_of_update. Which in this case should not log that at all.

Thanks in advance!

**Comment by stephenroller** [comment]
tl;dr: everything is okay.

I appreciate the filing. The metric is misleading. In this case, `num_updates` is the TOTAL NUMBER of completed SGD steps we have completed. It is monotonically increasing, but should never increase during validation/test time.

As a point of comparison, run `eval_model.py -t yourtask -mf yourmodelfile -ltim 2`. You'll notice that `num_updates` is printed on every line, but never changes.

I find it useful to log because it gives me a point of comparison when I'm running sweeps to see if two models have had similar number of updates.

**Comment by stephenroller** [comment]
I'll discuss with my colleagues and decide if we want to keep the reporting.

**Comment by newpro** [comment]
Cool thanks for your reply and the quick update! :laughing:

**Comment by stephenroller** [comment]
To remove ambiguity, we'll rename it to `total_train_updates`.
